### DISCUSSION TITLE
CorsixTH 0.65

### DISCUSSION THREAD

**Discussion by TheCycoONE** [CorsixTH/CorsixTH#2353] [Team Posts]
I would like to release next weekend (2021-06-13). @colinjmatt and @rmnvgr is that good for you, and when would you need a postdated appdata / release tag?

**Comment by TheCycoONE** [comment]
And @rmnvgr

**Comment by TheCycoONE** [comment]
(I shouldn't have mentioned names - anyone interested in coordinating with the release)

**Comment by alexandre-lavoie** [comment]
Nixpkgs is still under review... I think I'll be able to commit to release with 0.65 though!

**Comment by colinjmatt** [comment]
I'm happy with that date. I can handle ~24 hours notice as a reminder so I don't miss the release date.

**Comment by rmnvgr** [comment]
That's fine with me, the Flatpak can be updated whenever we need.

Is it possible to include the release notes in the appdata? They can be added in HTML in a [`description` tag](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-description) in the [`release` tag](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-releases).

**Comment by TheCycoONE** [comment]
Is there a way to test this before release? I'll go through the standard PR process so you can hopefully catch any errors.

**Comment by rmnvgr** [comment]
Ping me in the PR, I'll [check it locally](https://github.com/flathub/flathub/wiki/AppData-Guidelines#use-flathubs-appstream-util) to make sure everything is fine.

**Comment by TheCycoONE** [comment]
There is now a v0.65 tag in the repo and a draft release on GitHub. The plan is not to release until the 13th; binaries can be built and attached to the draft release any time between now and then.

I think it would be appropriate to add a flathub link to the release page as well - is it possible to link to a particular version?

**Comment by rmnvgr** [comment]
The way Flatpak works, the "Install" button on Flathub actually sets up the CorsixTH Flatpak repository on the user's system, and will automatically download the latest version available. So it's not really possible to have a link to a particular version.

**Comment by colinjmatt** [comment]
MacOS .dmg has been uploaded.

**Comment by tim77** [comment]
I've built 0.65 for Fedora and currently in `testing` repos so users have margin and time to test before it goes to `stable` - https://bodhi.fedoraproject.org/updates/?packages=corsix-th

Also this helps to detect some metainfo nuance. This affect only "old" distros which contains old `appstream-util`:
```
/builddir/build/BUILDROOT/corsix-th-0.65-1.fc33.x86_64/usr/share/metainfo/com.corsixth.corsixth.metainfo.xml: FAILED:
? attribute-invalid     : <release> timestamp is in the future
Validation of files failed
```

This is not a big issue and i'll build it for f33 later. This is even better to not hurry up for "oldstable" Fedora.

**Comment by TheCycoONE** [comment]
Unfortunately we are going to delay the final release. #1924 is a severe regression that is likely to come up during normal play.

**Comment by TheCycoONE** [comment]
Q1. Would it be a problem for anyone if I moved the v0.65 tag?
Q2. How much notice do you need when the fix lands?

Still waiting for a review on the PR https://github.com/CorsixTH/CorsixTH/pull/1925

@tim77 Putting it in testing before the official release might have been premature?

**Comment by tim77** [comment]
> @tim77 Putting it in testing before the official release might have been premature?

That's exactly what Rawhide and Testing branch for. I'll disable auto pushing to Stable until we will be ready.

**Comment by colinjmatt** [comment]
Happy to make another MacOS package as and when. About a day is the most notice I'll need, but I'll generally have it up quicker.

**Comment by rmnvgr** [comment]
I'll build the Flatpak as soon as 0.65 is ready.

**Comment by TheCycoONE** [comment]
There is a new v0.65 tag now, this time postdated for the 19th. I've deleted all the existing assets from the release draft page.

**Comment by tim77** [comment]
@TheCycoONE force-pushing could be issue sometimes for Linux distros and causing some confusing. Discouraging in general. IMO better next time `0.65.1` or `0.66-beta`.

**Comment by TheCycoONE** [comment]
@tim77 that's exactly what I asked about above:
"Q1. Would it be a problem for anyone if I moved the v0.65 tag?"

**Comment by TheCycoONE** [comment]
@rmnvgr You can publish any time, I'm about to publish the page here, but will hold off publicity until it's ready. I tried to get the flathubbot to rebuild the package, but either I don't have permission or I don't know what I'm doing.

**Comment by rmnvgr** [comment]
@TheCycoONE The 0.65 Flatpak build is ready, it will be automatically published on Flathub in a few hours, but I can force it anytime.

And the bot incantation is `bot, build` ;)

**Comment by TheCycoONE** [comment]
Ok, spiffy. In the live chat I mentioned that we're holding for a couple hours to see if anyone notices issues with any of the builds, and then assuming not I'll start making announcements. I'll make sure it's there before I announce.

**Comment by TheCycoONE** [comment]
@rmnvgr it's been 3 hours - let's force it. (BTW how is that done?)

**Comment by rmnvgr** [comment]
Done. You can go to https://flathub.org/builds/ to manage builds.

**Comment by TheCycoONE** [comment]
Thank you everyone, we are released!

**Comment by tim77** [comment]
[Built for Fedora](https://bodhi.fedoraproject.org/updates/?packages=corsix-th).
